src/Sentry/Protocol/Envelopes/Envelope.cs

```diff
@@ -445,6 +445,24 @@ internal static Envelope FromClientReport(ClientReport clientReport)
         return new Envelope(header, items);
     }
 
+    /// <summary>
+    /// Creates an envelope that contains only an attachment for an existing event.
+    /// </summary>
+    internal static Envelope FromAttachment(SentryId eventId, SentryAttachment attachment, IDiagnosticLogger? logger = null)
+    {
+        var header = CreateHeader(eventId);
+        var items = new List<EnvelopeItem>();
+
+        // Safety check, in case the user forcefully added a null attachment.
+        if (attachment.IsNull())
```

Comment:
question: null-check required?

This is an `internal` API, and `sentry-unity` (at least during build-time) does use nullable reference types (or at least `#nullable enable warnings`), right? So we should get warnings/errors if possible null.

Well ... on second thought ... should we end up making this API `public` in the future, we could potentially forget adding that check. Perhaps we actually *should* keep it.

Reply:
I've moved the check out into the hub.